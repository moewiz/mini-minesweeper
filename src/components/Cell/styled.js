import styled, { css } from "styled-components";

export const CellWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  border: 1px solid;
  border-color: #000;
  background: ${props => (props.isOpen ? "#e4e4e4" : "#fff")};
  font-weight: 600;
  ${props =>
    props.isMine &&
    props.isOpen &&
    css`
      position: relative;
      &:after {
        content: "";
        position: absolute;
        background: #24292e !important;
        border: 1px solid #24292e;
        border-radius: 50%;
        top: 4px;
        left: 4px;
        width: 20px;
        height: 20px;
      }
    `}
  &:hover {
    background: #e4e4e4;
  }
`;
