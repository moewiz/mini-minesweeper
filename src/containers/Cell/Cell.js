import React from "react";
import { Cell } from "../../components";
import { GAME_STATUS } from "../../constants/game";

const CellContainer = props => {
  const openCell = cell => {
    console.log("click on cell", cell);
  };
  const gameStatus = GAME_STATUS.WIN;

  return <Cell {...props} openCell={openCell} gameStatus={gameStatus} />;
};

export default CellContainer;
