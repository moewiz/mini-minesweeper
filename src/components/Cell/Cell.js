import React from "react";
import PropTypes from "prop-types";
import { CellWrapper } from "./styled";
import { GAME_STATUS } from "../../constants/game";

const Cell = ({ cell, openCell, gameStatus }) => {
  const isOpen =
    cell.isOpen ||
    gameStatus === GAME_STATUS.WIN ||
    gameStatus === GAME_STATUS.LOSE;

  return (
    <CellWrapper
      isOpen={isOpen}
      isMine={cell.minesAround === -1}
      onClick={() => openCell(cell)}
    >
      {isOpen && cell.minesAround > 0 && cell.minesAround}
    </CellWrapper>
  );
};

Cell.propTypes = {
  cell: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired,
    minesAround: PropTypes.number.isRequired,
    isOpen: PropTypes.bool.isRequired
  }),
  openCell: PropTypes.func.isRequired,
  gameStatus: PropTypes.number.isRequired
};

export default Cell;
