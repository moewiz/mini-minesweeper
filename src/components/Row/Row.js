import React from "react";
import PropTypes from "prop-types";
import { Cell } from "../../containers";
import { RowWrapper } from "./styled";

const Row = ({ row }) => {
  return (
    <RowWrapper>
      {row.map((cell, index) => (
        <Cell cell={cell} key={index} />
      ))}
    </RowWrapper>
  );
};

Row.propTypes = {
  row: PropTypes.arrayOf(
    PropTypes.shape({
      minesAround: PropTypes.number,
      isOpen: PropTypes.bool.isRequired,
      x: PropTypes.number.isRequired,
      y: PropTypes.number.isRequired
    })
  ).isRequired
};

export default Row;
