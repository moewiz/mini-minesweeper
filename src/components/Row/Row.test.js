import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import Row from "./Row";
import Cell from "../Cell";
import configureStore from "../../store";
const { store } = configureStore();

describe("<Row />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const Component = mount(
    <Provider store={store}>
      <Row
        row={[
          { x: 0, y: 0, minesAround: 0, isOpen: false },
          { x: 0, y: 1, minesAround: 0, isOpen: false },
          { x: 0, y: 2, minesAround: 0, isOpen: false },
          { x: 0, y: 3, minesAround: 0, isOpen: false }
        ]}
      />
    </Provider>
  );

  it("should render without throwing an error", () => {
    expect(Component.length).toBe(1);
  });

  it("should match snapshot", () => {
    expect(Component).toMatchSnapshot();
  });

  it("should render 4 Cell component", () => {
    const Cells = Component.find(Cell);
    expect(Cells.length).toBe(4);
  });
});
