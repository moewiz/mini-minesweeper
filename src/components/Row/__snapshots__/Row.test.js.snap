// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Row
    row={
      Array [
        Object {
          "isOpen": false,
          "minesAround": 0,
          "x": 0,
          "y": 0,
        },
        Object {
          "isOpen": false,
          "minesAround": 0,
          "x": 0,
          "y": 1,
        },
        Object {
          "isOpen": false,
          "minesAround": 0,
          "x": 0,
          "y": 2,
        },
        Object {
          "isOpen": false,
          "minesAround": 0,
          "x": 0,
          "y": 3,
        },
      ]
    }
  >
    <styled.div>
      <div
        className="sc-fznKkj iOOVlW"
      >
        <CellContainer
          cell={
            Object {
              "isOpen": false,
              "minesAround": 0,
              "x": 0,
              "y": 0,
            }
          }
          key="0"
        >
          <Cell
            cell={
              Object {
                "isOpen": false,
                "minesAround": 0,
                "x": 0,
                "y": 0,
              }
            }
            gameStatus={0}
            openCell={[Function]}
          >
            <styled.div
              isMine={false}
              isOpen={false}
              onClick={[Function]}
            >
              <div
                className="sc-fzozJi cgtCDw"
                onClick={[Function]}
              />
            </styled.div>
          </Cell>
        </CellContainer>
        <CellContainer
          cell={
            Object {
              "isOpen": false,
              "minesAround": 0,
              "x": 0,
              "y": 1,
            }
          }
          key="1"
        >
          <Cell
            cell={
              Object {
                "isOpen": false,
                "minesAround": 0,
                "x": 0,
                "y": 1,
              }
            }
            gameStatus={0}
            openCell={[Function]}
          >
            <styled.div
              isMine={false}
              isOpen={false}
              onClick={[Function]}
            >
              <div
                className="sc-fzozJi cgtCDw"
                onClick={[Function]}
              />
            </styled.div>
          </Cell>
        </CellContainer>
        <CellContainer
          cell={
            Object {
              "isOpen": false,
              "minesAround": 0,
              "x": 0,
              "y": 2,
            }
          }
          key="2"
        >
          <Cell
            cell={
              Object {
                "isOpen": false,
                "minesAround": 0,
                "x": 0,
                "y": 2,
              }
            }
            gameStatus={0}
            openCell={[Function]}
          >
            <styled.div
              isMine={false}
              isOpen={false}
              onClick={[Function]}
            >
              <div
                className="sc-fzozJi cgtCDw"
                onClick={[Function]}
              />
            </styled.div>
          </Cell>
        </CellContainer>
        <CellContainer
          cell={
            Object {
              "isOpen": false,
              "minesAround": 0,
              "x": 0,
              "y": 3,
            }
          }
          key="3"
        >
          <Cell
            cell={
              Object {
                "isOpen": false,
                "minesAround": 0,
                "x": 0,
                "y": 3,
              }
            }
            gameStatus={0}
            openCell={[Function]}
          >
            <styled.div
              isMine={false}
              isOpen={false}
              onClick={[Function]}
            >
              <div
                className="sc-fzozJi cgtCDw"
                onClick={[Function]}
              />
            </styled.div>
          </Cell>
        </CellContainer>
      </div>
    </styled.div>
  </Row>
</Provider>
`;
